elastictransport/elastictransport.go

```diff
@@ -66,11 +66,11 @@ type Config struct {
 	Header http.Header
 	CACert []byte
 
-	RetryOnStatus        []int
-	DisableRetry         bool
-	EnableRetryOnTimeout bool
-	MaxRetries           int
-	RetryBackoff         func(attempt int) time.Duration
+	RetryOnStatus       []int
+	DisableRetry        bool
+	DisableRetryOnError func(http.Request, error) bool
```

Comment:
```suggestion
	// DisableRetry disables retrying requests.
	//
	// If DisableRetry is true, then RetryOnStatus, RetryOnError, MaxRetries, and RetryBackoff will be ignored.
	DisableRetry        bool

	// RetryOnStatus holds an optional list of HTTP response status codes that should trigger a retry.
	//
	// If RetryOnStatus is nil, then the defaults will be used:
	// 502 (Bad Gateway), 503 (Service Unavailable), 504 (Gateway Timeout).
	RetryOnStatus       []int
	
	// RetryOnError holds an optional function that will be called when a request fails due to an
	// HTTP transport error, to indicate whether the request should be retried, e.g. timeouts.
	RetryOnError func(*http.Request, error) bool
```

I think it might be a little easier to understand if the function were inverted to be `RetryOnError`. This would also be more in line with `RetryOnStatus`.

Perhaps also change that `http.Request` to `*http.Request` to avoid copying? The struct itself contains pointer fields, so there's limited value in copying the struct in terms of safety. I'd just document that the function should not modify the request.

Reply:
I was hesitant on the pointer but you are right about the internal fields.

Definitely agree on `RetryOnError`, I'll update the PR accordingly!